### PR TITLE
ALSA/ASoC: hda: move SPIB/DRMS functionality from ext layer

### PR DESCRIPTION
**Pull request by plbossart**
The SPIB and DRMS capabilities are orthogonal to the DSP enablement and can be used whether the stream is coupled or not.

The existing code partitioning makes limited sense, the capabilities are parsed at the sound/hda level but helpers are located in sound/hda/ext.

This patch moves all the SPIB/DRMS functionality to the sound/hda layer. This reduces the complexity of the sound/hda/ext layer which is now limited to handling the multi-link extensions and stream coupling/decoupling helpers.

Note that this is an iso-functionality code move and rename, the HDaudio legacy driver would need additional changes to make use of these capabilities.

Signed-off-by: Pierre-Louis Bossart <pierre-louis.bossart@linux.intel.com>

### CODE REVIEW

File: include/sound/hdaudio.h [hunk 1]
```diff
@@ -495,6 +495,15 @@ static inline u16 snd_hdac_reg_readw(struct hdac_bus *bus, void __iomem *addr)
 	snd_hdac_chip_writeb(chip, reg, \
 			     (snd_hdac_chip_readb(chip, reg) & ~(mask)) | (val))
 
+/* update register macro */
+#define snd_hdac_updatel(addr, reg, mask, val)		\
+	writel(((readl(addr + reg) & ~(mask)) | (val)), \
+		addr + reg)
```
Comment (ujfalusi) [inline include/sound/hdaudio.h]: this can comfortably fit in a single line

Comment (ujfalusi) [reply include/sound/hdaudio.h]: I mean in two lines, that is.

**Review by ujfalusi — COMMENTED** [review]
@plbossart, great, only one nit-pick

File: include/sound/hdaudio.h (hunk 1 above)
Comment (plbossart) [reply include/sound/hdaudio.h]: done

**Review by ranj063 — APPROVED** [review]

**Comment by plbossart** [comment]
@bardliao @ujfalusi can you re-check the update, I'd like to send this upstream tomorrow. Thanks!

**Review by bardliao — APPROVED** [review]

**Pull request merged**